test/test_common/network_utility.h

```diff
@@ -180,8 +180,11 @@ Api::IoCallUint64Result readFromSocket(IoHandle& handle, const Address::Instance
  */
 class UdpSyncPeer {
 public:
+  // Create a peer that listens on the given version of the loopback address.
   UdpSyncPeer(Network::Address::IpVersion version);
 
+  // Create a peer that listens on the given address:port.
+  UdpSyncPeer(Network::Address::InstanceConstSharedPtr address);
```

Comment:
For tests we can't bind to a particular port. Similar to the UDP statsd tests, you should create a target which will bind to port 0 and then determine that actual port, and then feed that into your test.